### DISCUSSION TITLE
Present a permutation of image stimuli from source folder

### DISCUSSION THREAD

**Discussion by ThetaS** [bonsai-rx/bonsai#1560] [Q&A]
Hi all,

I ran into a problem not being able to present a permutation of images from a folder. Similar topics are discussed in #1237 and others but the difference is that we need to present in pseudo random that all images should be presented with the same number of repeats by the end. I have problems using the "concat" node and finding a way to iterating through all the loaded texture for this purpose. Attached is the source file I am modifying on. Thanks!

[image_stim.zip](https://github.com/bonsai-rx/bonsai/files/12644565/image_stim.zip)

**Comment by bruno-f-cruz** [comment]
Not sure I got what you need to do exactly, but let me know if this is not the direction:

### Load and store the sequence of strings in a `ReplaySubject`
![image](https://github.com/bonsai-rx/bonsai/assets/7049351/1679de10-e391-4388-9e81-0e5f7f6eeab7)

### Repeat your sequence N times using `RepeatCount`

![image](https://github.com/bonsai-rx/bonsai/assets/7049351/ff50da95-3e9a-459b-9c40-e47978464b19)

If you want to use the same permutation on each "loop" keep the `Permutation` as is, if you want to use a different one each time, move it to after the subscription to `MyFiles`.

[1560.zip](https://github.com/bonsai-rx/bonsai/files/12648114/1560.zip)

**Comment by ThetaS** [comment]
Hi Bruno @bruno-f-cruz ,

You are on the right track. We want to present 400 random trials while ensuring each image gets presented 4 repeats in pseudo random order. I guess it might work to just move the permutation node after "MyFile" subscription


Technical issues that are currently unsolved are: 1) no intermediate blank period was shown between stimulus. 2) take until fail to specify the onset period. Like here in the "Create Observable":

![image](https://github.com/bonsai-rx/bonsai/assets/32575952/1fa78567-ee45-40bf-8ee3-2eeaacf80c69)

As for testing, I set the due time to 10 sec between stimulus but instead of present one image at a time for 10 sec gaps, it flashes all at once at the start. This still happens after I added a 3 sec Shader delay. I don't quite get why this happens.
